File: C++/Problems/Problems/problem_2839.cpp
#include "problem_2839.h"

bool problem_2839::test() {
    std::string s1 = "abcd";
    std::string s2 = "cdab";

    bool expected = true;

    auto result = canBeEqual(s1, s2);

    return result == expected;
}

bool problem_2839::canBeEqual(std::string s1, std::string s2) {
    int n = s1.size();
    for (int i = 0; i < n - 2; ++i)
    {
        if ((s1[i] != s2[i] || s1[i + 2] != s2[i + 2]) && (s1[i] != s2[i + 2] || s1[i + 2] != s2[i]))
            return false;
    }

    return true;
}

File: Python/Problems/Problems/problem2839.py
import heapq
import math
import sys

from utils import print1DArray, print2DArray
from typing import List, Set, Optional
from collections import Counter

from .problem import Problem

class Problem2839(Problem):
    def test(self) -> bool:
        s1 = "abcd"
        s2 = "cdab"

        expected = True

        result = self.canBeEqual(s1, s2)

        return result == expected

    def canBeEqual(self, s1: str, s2: str) -> bool:
        n = len(s1)
        for i in range(n - 2):
            if (s1[i] != s2[i] or s1[i + 2] != s2[i + 2]) and (s1[i] != s2[i + 2] or s1[i + 2] != s2[i]):
                return False

        return True